src/ape/managers/_contractscache.py

```diff
@@ -561,41 +561,26 @@ def get(
 
             return contract_type
 
-        else:
-            # Contract is not cached yet. Check broader sources, such as an explorer.
-            if not proxy_info and detect_proxy:
-                # Proxy info not provided. Attempt to detect.
-                if not (proxy_info := self.proxy_infos[address_key]):
-                    if proxy_info := self.provider.network.ecosystem.get_proxy_info(address_key):
-                        self.proxy_infos[address_key] = proxy_info
-
-            if proxy_info:
-                # Contract is a proxy (either was detected or provided).
-                implementation_contract_type = self.get(proxy_info.target, default=default)
-                proxy_contract_type = (
-                    self._get_contract_type_from_explorer(address_key)
-                    if fetch_from_explorer
-                    else None
-                )
-                if proxy_contract_type is not None and implementation_contract_type is not None:
-                    combined_contract = _get_combined_contract_type(
-                        proxy_contract_type, proxy_info, implementation_contract_type
-                    )
-                    self.contract_types[address_key] = combined_contract
-                    return combined_contract
-
-                elif implementation_contract_type is not None:
-                    contract_type_to_cache = implementation_contract_type
-                    self.contract_types[address_key] = implementation_contract_type
-                    return contract_type_to_cache
-
-                elif proxy_contract_type is not None:
-                    self.contract_types[address_key] = proxy_contract_type
-                    return proxy_contract_type
-
-            # Also gets cached to disk for faster lookup next time.
-            if fetch_from_explorer:
-                contract_type = self._get_contract_type_from_explorer(address_key)
+        # Contract is not cached yet. Check broader sources, such as an explorer.
+        if not proxy_info and detect_proxy:
+            # Proxy info not provided. Attempt to detect.
+            if not (proxy_info := self.proxy_infos[address_key]):
+                if proxy_info := self.provider.network.ecosystem.get_proxy_info(address_key):
+                    self.proxy_infos[address_key] = proxy_info
+
+        if proxy_info:
+            # Contract is a proxy (either was detected or provided).
+            if impl_contract := self._get_proxy_contract_type(
```

Comment:
This can be one of 4 things:
1. combined proxy + impl abi
2. impl-only abi (for things like simple forwarder proxies)
3. proxy-only abi (for things like upgradable proxies that don't have impls, or unverified impl)
4. undeterminable contract_type (eitehr impl or proxy)

I think should add a note about this, and consider calling this variable like `proxy_contract_type` instead (maybe?)